src/web/app/user-join-page.component.spec.ts

```diff
@@ -286,26 +295,69 @@ describe('UserJoinPageComponent creating account', () => {
     expect(navSpy).toHaveBeenLastCalledWith(expect.anything(), '/web/instructor/home');
   });
 
-  it('should stop loading and show error message if 404 is returned when creating new account', () => {
-    jest.spyOn(authService, 'getAuthUser').mockReturnValue(of({
-      user: {
-        id: 'user',
-        isAdmin: false,
-        isInstructor: false,
-        isStudent: false,
-        isMaintainer: false,
-      },
-      masquerade: false,
-    }));
-    jest.spyOn(courseService, 'getJoinCourseStatus').mockReturnValue(throwError({
-      status: 404,
-    }));
+  describe('should stop loading and show error message', () => {
+    beforeEach(() => {
```

Comment:
I'm so sorry, I missed this earlier. The additional testcases added are in the wrong section. This describes groups cases for creating account (because the query params for these cases are different from the join course cases).

The two new cases 4xx and 5xx should be moved to line 160 (replacing TC 'should stop loading and show error message if 404 is returned').